application/tests/unit/common/components/LdapTest.php

```diff
@@ -47,8 +44,8 @@ public function testRemoveAttributesOnSet()
         /*
          * Get user before change to ensure presence of attributes to be removed
          */
-        /** @var \Adldap\Models\Entry $user */
-        $user = $ldap->ldapProvider->search()
+        /** @var \LdapRecord\Models\OpenLDAP\Entry $user */
+        $user = $ldap->ldapConnection->query()
```

Comment:
A double -> is ugly, in my opinion. Also, "slower" access. Having an $ldapConnection = $ldap->ldapConnection; would also be reusable below.

Reply:
I don't think it is reusable. I think it needs to be set again after every `$ldap->connect()` call.